subsys/bluetooth/host/hci_core.c

```diff
@@ -2603,7 +2603,7 @@ static void le_read_buffer_size_complete(struct net_buf *buf)
 
 #if defined(CONFIG_BT_CONN)
 	bt_dev.le.acl_mtu = sys_le16_to_cpu(rp->le_max_len);
-	if (!bt_dev.le.acl_mtu) {
+	if (!bt_dev.le.acl_mtu && !rp->le_max_num) {
```

Comment:
A couple of notes:
- It should be `||`, not `&&`. I.e. if either parameter is zero you ignore the command response.
- Since you already assigned to `bt_dev.le.acl_mtu` before this test, it means that you didn't actually completely ignore the command response. It might be better to create a local `uint16_t acl_mtu` variable, and only after you've confirmed that you have non-zero command parameters do you assign `bt_dev.le.acl_mtu = acl_mtu;`